File: CodingTest/Programmers/level1/WeirdString.java
package CodingTest.Programmers.level1;

public class WeirdString {
    public static void main(String[] args){
        String s = " AAA A A A  ";
        System.out.println(solution(s));
    }

    public static String solution(String s){
        String answer="";
        String ss = s.replaceAll(" ", "_");
        String[] temp = ss.split("_",-1);
        String tempS ="";
        for(int i=0; i<temp.length; i++){
            if(temp[i].isEmpty()){
                answer+=" ";
                continue;
            }
            for(int j=0; j<temp[i].length(); j++){
                if(j%2==0){
                    tempS = Character.toString(temp[i].charAt(j));
                    answer += tempS.toUpperCase();
                    tempS ="";
                }
                else{
                    tempS = Character.toString(temp[i].charAt(j));
                    answer += tempS.toLowerCase();
                    tempS="";
                }
            }
            if(i!=temp.length-1){
                answer+=" ";
            }
        }
        
        if(answer.length()>s.length()){
            return answer.substring(0, answer.length()-1);
        }

        return answer;
    }
}

File: CodingTest/Programmers/level1/WeirdString.js
function solution(s) {
    var count = 0;
    var answer = "";
    
    for(var i=0; i<s.length; ++i){
        if(s[i]==" "){
            answer += " ";
            count = 0;
            continue;
        }    
        
        if(count%2==0) answer += s[i].toUpperCase();
        else answer += s[i].toLowerCase();
        
        count++;
    }
    
    return answer;
}